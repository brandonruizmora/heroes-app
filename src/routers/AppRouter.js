import React, { useContext } from 'react';
import {
    BrowserRouter,
    Switch,
} from 'react-router-dom';
import { AuthContext } from '../auth/AuthContext';
import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

export const AppRouter = () => {

    const { user: { logged } } = useContext(AuthContext);

    return (
        <BrowserRouter basename="/">
            <div>
                <Switch>
                    <PublicRoute
                        exact
                        path="/login"
                        component={LoginScreen}
                        isAuthenticated={logged}
                    />
                    <PrivateRoute
                        path="/"
                        component={DashboardRoutes}
                        isAuthenticated={logged}
                    />
                </Switch>
            </div>
        </BrowserRouter>
    )
}
