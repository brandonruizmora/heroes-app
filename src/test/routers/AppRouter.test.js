import '../../setupTests';
import { mount } from 'enzyme';
import { AppRouter } from '../../routers/AppRouter';
import { AuthContext } from '../../auth/AuthContext';

describe('Pruebas en <AppRouter />', () => {

    test('Debe de mostrar login cuando no esta autenticado', () => {
        
        const contextValue = {
            user: {
                logged: false
            }
        }

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />

            </AuthContext.Provider>
        );

        expect( wrapper ).toMatchSnapshot();

    });

    test('Debe de mostrar componente marvel si esta autenticado', () => {
        
        const contextValue = {
            user: {
                name: 'Brandon',
                logged: true
            }
        }

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />

            </AuthContext.Provider>
        );

        expect( wrapper ).toMatchSnapshot();

    });
    
    
})
