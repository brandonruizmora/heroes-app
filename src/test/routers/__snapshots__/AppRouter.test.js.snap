// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> Debe de mostrar componente marvel si esta autenticado 1`] = `
<div>
  Array [
    <nav
      className="navbar navbar-expand-lg navbar-dark bg-dark"
    >
      <div
        className="container-fluid"
      >
        <a
          className="navbar-brand"
          href="#/"
          onClick={[Function]}
        >
          Asociaciones
        </a>
        <button
          aria-controls="navbarSupportedContent"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-bs-target="#navbarSupportedContent"
          data-bs-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarSupportedContent"
        >
          <div
            className="navbar-nav me-auto mb-2 mb-lg-0"
          >
            <a
              aria-current="page"
              className="nav-item nav-link active"
              href="#/marvel"
              onClick={[Function]}
              style={Object {}}
            >
              Marvel
            </a>
            <a
              aria-current={null}
              className="nav-item nav-link"
              href="#/dc"
              onClick={[Function]}
            >
              DC
            </a>
            <a
              aria-current={null}
              className="nav-item nav-link"
              href="#/Search"
              onClick={[Function]}
            >
              Search
            </a>
          </div>
          <form
            className="d-flex"
          >
            <span
              className="align-self-center text-info me-3"
            >
              Brandon
            </span>
            <button
              className="nav-link btn text-white btn-outline-danger"
              onClick={[Function]}
            >
              Logout
            </button>
          </form>
        </div>
      </div>
    </nav>,
    <div
      className="container"
    >
      <div>
        <h1
          className="fw-bold text-yellow text-shadow-dark"
        >
          Marvel Screen
        </h1>
        <hr />
        <div
          className="row row-cols-1 row-cols-md-2 row-cols-lg-4 row-cols-xl-8 g-8 animate__animated animate__fadeIn"
        >
          <div
            className="card m-3"
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="marvel-spider"
                  className="img-fluid rounded-start"
                  src="./assets/heroes/marvel-spider.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Spider Man
                  </h5>
                  <p
                    className="card-text"
                  >
                    Peter Parker
                  </p>
                  <p
                    className="card-text"
                  />
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Amazing Fantasy #15
                    </small>
                  </p>
                  <a
                    href="#/hero/marvel-spider"
                    onClick={[Function]}
                  >
                    Más
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card m-3"
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="marvel-captain"
                  className="img-fluid rounded-start"
                  src="./assets/heroes/marvel-captain.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Captain America
                  </h5>
                  <p
                    className="card-text"
                  >
                    Steve Rogers
                  </p>
                  <p
                    className="card-text"
                  />
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Captain America Comics #1
                    </small>
                  </p>
                  <a
                    href="#/hero/marvel-captain"
                    onClick={[Function]}
                  >
                    Más
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card m-3"
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="marvel-iron"
                  className="img-fluid rounded-start"
                  src="./assets/heroes/marvel-iron.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Iron Man
                  </h5>
                  <p
                    className="card-text"
                  >
                    Tony Stark
                  </p>
                  <p
                    className="card-text"
                  />
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Tales of Suspense #39
                    </small>
                  </p>
                  <a
                    href="#/hero/marvel-iron"
                    onClick={[Function]}
                  >
                    Más
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card m-3"
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="marvel-thor"
                  className="img-fluid rounded-start"
                  src="./assets/heroes/marvel-thor.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Thor
                  </h5>
                  <p
                    className="card-text"
                  >
                    Thor Odinson
                  </p>
                  <p
                    className="card-text"
                  />
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Journey into Myster #83
                    </small>
                  </p>
                  <a
                    href="#/hero/marvel-thor"
                    onClick={[Function]}
                  >
                    Más
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card m-3"
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="marvel-hulk"
                  className="img-fluid rounded-start"
                  src="./assets/heroes/marvel-hulk.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Hulk
                  </h5>
                  <p
                    className="card-text"
                  >
                    Bruce Banner
                  </p>
                  <p
                    className="card-text"
                  />
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      The Incredible Hulk #1
                    </small>
                  </p>
                  <a
                    href="#/hero/marvel-hulk"
                    onClick={[Function]}
                  >
                    Más
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card m-3"
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="marvel-wolverine"
                  className="img-fluid rounded-start"
                  src="./assets/heroes/marvel-wolverine.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Wolverine
                  </h5>
                  <p
                    className="card-text"
                  >
                    James Howlett
                  </p>
                  <p
                    className="card-text"
                  />
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      The Incredible Hulk #180
                    </small>
                  </p>
                  <a
                    href="#/hero/marvel-wolverine"
                    onClick={[Function]}
                  >
                    Más
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card m-3"
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="marvel-daredevil"
                  className="img-fluid rounded-start"
                  src="./assets/heroes/marvel-daredevil.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Daredevil
                  </h5>
                  <p
                    className="card-text"
                  >
                    Matthew Michael Murdock
                  </p>
                  <p
                    className="card-text"
                  />
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Daredevil #1
                    </small>
                  </p>
                  <a
                    href="#/hero/marvel-daredevil"
                    onClick={[Function]}
                  >
                    Más
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card m-3"
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="marvel-hawkeye"
                  className="img-fluid rounded-start"
                  src="./assets/heroes/marvel-hawkeye.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Hawkeye
                  </h5>
                  <p
                    className="card-text"
                  >
                    Clinton Francis Barton
                  </p>
                  <p
                    className="card-text"
                  />
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Tales of Suspense #57
                    </small>
                  </p>
                  <a
                    href="#/hero/marvel-hawkeye"
                    onClick={[Function]}
                  >
                    Más
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card m-3"
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="marvel-cyclops"
                  className="img-fluid rounded-start"
                  src="./assets/heroes/marvel-cyclops.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Cyclops
                  </h5>
                  <p
                    className="card-text"
                  >
                    Scott Summers
                  </p>
                  <p
                    className="card-text"
                  />
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      X-Men #1
                    </small>
                  </p>
                  <a
                    href="#/hero/marvel-cyclops"
                    onClick={[Function]}
                  >
                    Más
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card m-3"
          >
            <div
              className="row g-0"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="marvel-silver"
                  className="img-fluid rounded-start"
                  src="./assets/heroes/marvel-silver.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Silver Surfer
                  </h5>
                  <p
                    className="card-text"
                  >
                    Norrin Radd
                  </p>
                  <p
                    className="card-text"
                  />
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      The Fantastic Four #48
                    </small>
                  </p>
                  <a
                    href="#/hero/marvel-silver"
                    onClick={[Function]}
                  >
                    Más
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>,
  ]
</div>
`;

exports[`Pruebas en <AppRouter /> Debe de mostrar login cuando no esta autenticado 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="fill d-flex align-items-center justify-content-center"
    >
      <div
        className="card w-50 h-50"
      >
        <div
          className="card-header"
        >
          <h1>
            Login Screen
          </h1>
        </div>
        <div
          className="card-body d-flex align-items-center justify-content-center"
        >
          <div
            className="row"
          >
            <div
              className="col-12"
            >
              <input
                autoComplete="off"
                className="form-control"
                maxLength="20"
                name="userName"
                onChange={[Function]}
                placeholder="Write an username..."
                type="text"
                value=""
              />
              <small
                className="fw-light fst-italic float-end me-1"
              >
                0
                 / 20
              </small>
            </div>
            <div
              className="col-12"
            >
              <div
                className="d-grid gap-1 w-100"
              >
                <button
                  className="btn btn-dark mt-4"
                  onClick={[Function]}
                >
                  Login
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
